### PR TITLE
JAMES-2470 REST SMTP container management imrpovements

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] Results :
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] Failed tests: 
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   CassandraMailboxDAOTest Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.17.0.1:48577 (com.datastax.driver.core.exceptions.OperationTimedOutException: [/172.17.0.1:48577] Operation timed out))
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	<no message> in java.lang.NullPointerException
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   CassandraMailboxRecentDAOTest Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.17.0.1:48577 (com.datastax.driver.core.exceptions.OperationTimedOutException: [/172.17.0.1:48577] Operation timed out))
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	<no message> in java.lang.NullPointerException
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   CassandraMessageIdDAOTest Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.17.0.1:48577 (com.datastax.driver.core.exceptions.OperationTimedOutException: [/172.17.0.1:48577] Operation timed out))
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	<no message> in java.lang.NullPointerException
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   CassandraMessageMapperTest Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.17.0.1:48577 (com.datastax.driver.core.exceptions.OperationTimedOutException: [/172.17.0.1:48577] Operation timed out))
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	<no message> in java.lang.NullPointerException
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   CassandraMessageWithAttachmentMapperTest Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.17.0.1:48577 (com.datastax.driver.core.exceptions.OperationTimedOutException: [/172.17.0.1:48577] Operation timed out))
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	<no message> in java.lang.NullPointerException
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.addingFlagToAMessageThatAlreadyHasThisFlagShouldResultInNoChange Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.copyInMailboxShouldSaveMessageInAnotherMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.copyInMailboxShouldThrowWhenMailboxDoesntExist Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.copyInMailboxShouldWorkWhenSavingTwoTimesWithSameMessageIdAndSameMailboxId Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countMessageShouldBeEmptyWhenDeleteMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	[/172.17.0.1:48577] Timed out waiting for server response
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.17.0.1:48577 (com.datastax.driver.core.exceptions.OperationTimedOutException: [/172.17.0.1:48577] Timed out waiting for server response))
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countMessageShouldReturnWhenCreateNewMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countUnseenMessageShouldBeEmptyWhenDeleteMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countUnseenMessageShouldBeEmptyWhenMessageIsSeen Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countUnseenMessageShouldNotTakeCareOfOtherFlagsUpdates Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countUnseenMessageShouldReturnWhenDeleteMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countUnseenMessageShouldReturnWhenMessageIsNotSeen Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countUnseenMessageShouldTakeCareOfMessagesMarkedAsRead Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.countUnseenMessageShouldTakeCareOfMessagesMarkedAsUnread Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deleteShouldDeleteAMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deleteShouldDeleteMessageIndicesWhenStoredInTwoMailboxes Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deleteShouldDeleteMessageIndicesWhenStoredTwoTimesInTheSameMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deleteShouldNotThrowWhenUnknownMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deleteWithMailboxIdsShouldDeleteIndicesWhenMailboxIdsContainsMultipleElements Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deleteWithMailboxIdsShouldDeleteOneIndexWhenMailboxIdsContainsOneElement Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deleteWithMailboxIdsShouldNotDeleteIndicesWhenMailboxIdsIsEmpty Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deletesShouldNotFailUponMissingMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deletesShouldOnlyRemoveConcernedMessages Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deletesShouldUpdateMessageCount Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.deletesShouldUpdateUnreadCount Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.findMailboxesShouldReturnEmptyWhenMessageDoesntExist Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.findMailboxesShouldReturnOneMailboxWhenMessageExistsInOneMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.findMailboxesShouldReturnTwoMailboxesWhenMessageExistsInTwoMailboxes Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.findShouldReturnEmptyWhenIdListIsEmpty Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.findShouldReturnMultipleMessagesWhenIdContainsMultiple Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.findShouldReturnMultipleMessagesWhenIdContainsMultipleInDifferentMailboxes Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.findShouldReturnOneMessageWhenIdListContainsOne Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.saveShouldSaveAMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.saveShouldSaveMessageInAnotherMailboxWhenMessageAlreadyInOneMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.saveShouldThrowWhenMailboxDoesntExist Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.saveShouldWorkWhenSavingTwoTimesWithSameMessageIdAndSameMailboxId Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldAddFlagsWhenAddUpdateMode Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldNotModifyFlagsWhenMailboxIdsIsEmpty Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldNotModifyModSeqWhenMailboxIdsIsEmpty Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldNotUpdateModSeqWhenNoop Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldReturnEmptyWhenMailboxIdsIsEmpty Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldReturnEmptyWhenMessageIdDoesntExist Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldReturnUpdatedFlagsWhenMessageIsInOneMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldReturnUpdatedFlagsWhenMessageIsInTwoMailboxes Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	[/172.17.0.1:48577] Connection has been closed
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldReturnUpdatedFlagsWhenNoop Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldReturnUpdatedFlagsWhenRemoveMode Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldReturnUpdatedFlagsWhenReplaceMode Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldUpdateFlagsWhenMessageIsInOneMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldUpdateFlagsWhenMessageIsInTwoMailboxes Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldUpdateMessageFlagsWhenRemoveMode Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldUpdateModSeqWhenMessageIsInOneMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldWorkWhenCalledOnDuplicatedMailbox Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldWorkWhenCalledOnFirstMessage Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldWorkWithConcurrencyWithAdd Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   MessageIdMapperTest.setFlagsShouldWorkWithConcurrencyWithRemove Multiple Failures (2 failures)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (no host was tried)
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] Tests in error: 
[826a536e703f8eb25fd4ec21e9a1272348cc30c6]   CassandraMessageIdMapperTest.findShouldReturnCorrectElementsWhenChunking » Completion
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] 
[826a536e703f8eb25fd4ec21e9a1272348cc30c6] Tests run: 636, Failures: 59, Errors: 1, Skipped: 0
```

File: server/testing/src/main/java/org/apache/james/utils/FakeSmtp.java [hunk 1]
```diff
@@ -105,4 +106,14 @@ private RequestSpecification requestSpecification() {
     public SwarmGenericContainer getContainer() {
         return container;
     }
+
+    public void clean() {
+        List<String> ids = given(requestSpecification(), RESPONSE_SPECIFICATION)
```
Comment (trantienduchn) [inline server/testing/src/main/java/org/apache/james/utils/FakeSmtp.java]: 👍
I have a look at fakesmtp repository, i think there is another way to truncate all emails which uses:
- `/api/email/latest` to retrieve the latest email id 
- `/api/email/purge/:id` to delete all emails older than an input id

Could you consider?

Comment (chibenwa) [reply server/testing/src/main/java/org/apache/james/utils/FakeSmtp.java]: > `/api/email/purge/:id` to delete all emails older than an input id


Isn't it earlier we would be interested in?

Anyway I'll have a look.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**